src/popup/router/pages/Retip.vue

```diff
@@ -34,69 +37,97 @@
 
 <script>
 import { mapGetters, mapState } from 'vuex';
-import BigNumber from 'bignumber.js';
-import tipping from 'aepp-raendom/src/utils/tippingContractUtil';
+import tipping from 'tipping-contract/util/tippingContractUtil';
 import { MAGNITUDE, calculateFee, TX_TYPES } from '../../utils/constants';
+import { convertToken } from '../../utils/helper';
 import deeplinkApi from '../../../mixins/deeplinkApi';
 import AmountSend from '../components/AmountSend';
 import UrlStatus from '../components/UrlStatus';
 import Button from '../components/Button';
+import BalanceInfo from '../components/BalanceInfo';
 
 export default {
   mixins: [deeplinkApi],
-  components: { AmountSend, UrlStatus, Button },
+  components: { AmountSend, UrlStatus, Button, BalanceInfo },
   data: () => ({
     tip: {},
     amount: null,
     amountError: false,
     loading: false,
     minCallFee: null,
+    tippingContract: null,
   }),
   computed: {
     ...mapGetters(['account', 'tippingSupported', 'minTipAmount', 'activeNetwork']),
-    ...mapState(['balance', 'tippingV1', 'sdk']),
+    ...mapState('fungibleTokens', ['selectedToken']),
+    ...mapState(['balance', 'tippingV1', 'tippingV2', 'sdk']),
     urlStatus() {
       return this.$store.getters['tipUrl/status'](this.tip.url);
     },
+    shouldUseV2Contract() {
+      return this.$route.query.id.includes('_v2') || this.$route.query.id.includes('_v3');
```

Comment:
Can you switch the contracts right from here?

Reply:
Also, should we support v2 contracts as far as they didn't come to production?

Reply:
We should support it. It is already done in the superhero-ui. The change would be just a matter of configuration when the time comes.

Reply:
Why the same contract is used for v2 and v3?

Reply:
V3 is only used for posts without tips. It would be tipped and re-tipped by V2 contract.